### PR TITLE
Migrate to Spine 0.8.0

### PR DESCRIPTION
**Pull request by dmdashenkov**
Update the storage impl. to core `v0.8.0`.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov PTAL

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public static void closeAll(Iterable<? extends AutoCloseable> closeables) {
+        checkNotNull(closeables);
+        try {
+            for (AutoCloseable closable : closeables) {
+                closable.close();
+            }
+        } catch (Exception e) {
+            throw new IllegalStateException(e);
```
Comment (alexander-yevsyukov) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java]: Why don't we try to continue closing other things that can be closed?

Comment (alexander-yevsyukov) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java]: Can we try collecting all the exceptions and then throwing them inside one aggregating instance passed to `IllegalStateException` as the cause?

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+package org.spine3.server.storage.jdbc.util;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A utility class for working with {@link AutoCloseable} instances.
+ *
+ * @author Dmytro Dashenkov.
+ */
+@SuppressWarnings("UtilityClass")
```
Comment (alexander-yevsyukov) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java]: I think we reduced the level of such things to “typo” in the `core-java` project. We can do it here to until we have an infrastructure for propagating IDEA settings automatically.

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/command/query/InsertCommandQuery.java [hunk 3]
```diff
@@ -20,13 +20,14 @@
 
 package org.spine3.server.storage.jdbc.command.query;
 
-import org.spine3.server.storage.CommandStorageRecord;
+import org.spine3.server.command.storage.CommandStorageRecord;
 
 import static org.spine3.server.storage.jdbc.command.query.CommandTable.COMMAND_COL;
```
Comment (alexander-yevsyukov) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/command/query/InsertCommandQuery.java]: Why don't we migrate to a newer version with `StorageField` and adopt the solution. Or, do you want to do it in a separate PR?

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java (hunk 1 above)
Comment (dmdashenkov) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java]: Yes, I'll provide this upgrade

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/command/query/InsertCommandQuery.java (hunk 3 above)
Comment (dmdashenkov) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/command/query/InsertCommandQuery.java]: I do want to have a separate PR for that. Now the latest version of `JDBC-storage` is `0.7.0` which is way outdated. To make it compatible with the release `v0.8.0` core, I've created this particular patch.

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov PTAL

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java [hunk 4]
```diff
@@ -27,20 +27,41 @@
...
         checkNotNull(closeables);
-        try {
-            for (AutoCloseable closable : closeables) {
+        Exception exception = null;
+        AutoCloseable faultyInstance = null;
+        for (AutoCloseable closable : closeables) {
+            try {
                 closable.close();
+            } catch (Exception e) {
+                if (exception == null) {
```
Comment (alexander-yevsyukov) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java]: This way we remember only the last one. But we need to remember all. Otherwise we lose information on all previous errors.

We need a custom exception, say `MultipleCloseException`, which would accept `Iterable<Exception>` to constructor. We gather all the exceptions, then create `MultipleCloseException` with them, and pass the as the cause to `IllegalStateException`.

Please pay attention to `toString()` of `MultipleCloseException`. It should print all the enclosed exceptions nicely.

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov PTAL

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/exception/MultipleCloseException.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    /**
+     * Describes all the given {@link Exception exceptions}.
+     *
+     * @return the description for all the exceptions
+     */
+    @Override
+    public String toString() {
+        @SuppressWarnings("StringBufferWithoutInitialCapacity")
```
Comment (alexander-yevsyukov) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/exception/MultipleCloseException.java]: Why `Throwables.getStackTraceAsString()` from Guava doesn't work for us here?

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        }
+        if (exceptions.isEmpty()) {
+            return;
+        }
+        if (exceptions.size() == 1) {
+            final Exception cause = exceptions.iterator()
+                                              .next();
+            throw new IllegalStateException(cause);
+        }
+        throw new MultipleCloseException(exceptions);
```
Comment (alexander-yevsyukov) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java]: In this case you'd need to catch for two exceptions, which is not convenient.

You need to throw ISE, but put different cause because you'll look into the cause anyway.

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov PTAL again.
